modules/nextflow/src/main/groovy/nextflow/util/ThreadPoolBuilder.groovy

```diff
@@ -141,7 +142,7 @@ class ThreadPoolBuilder {
         if( workQueue==null )
             workQueue = new LinkedBlockingQueue<>()
         if( rejectionPolicy==null )
-            rejectionPolicy = new ThreadPoolExecutor.CallerRunsPolicy()
+            rejectionPolicy = new ThreadPoolExecutor.AbortPolicy()
```

Comment:
This will change the default rejection policy to FinalizePool, PublishDir, FilePorter, and S3ThreadPool. Is this the intention? It could cause side effects. The error was in the S3ThreadPool, we could add a ThreadPoolManager.withRejectionPolicy method and set the AbortPolicy just for the S3 cases.